fix(node_manager): lowercase owner for discord

mazzi opened the pull request:
### Description

We use owner in lowercase to be able to compute scores accordingly on Discord.

Related https://github.com/maidsafe/dags-service/pull/7
Related https://github.com/maidsafe/safe_network/pull/2180

mazzi committed:
fix(node_manager): force lowercase discord username

maqi commented on sn_node_manager/src/add_services/mod.rs:
```diff
@@ -78,6 +78,11 @@ pub async fn add_node(
     }
 
+    let owner = match &options.owner {
+        Some(owner) => Some(owner.to_lowercase()),
```
shall we raise a warning here in case an `UpperCase` input detected ?

at least a `warning log` ?

mazzi replied:
Good idea.

mazzi committed:
fix(node_manager): adding warn to log in case the user has uppercase

jacderida commented:
Hey mazzi,

The node manager has a test suite. It would be nice if we could add a test case for this. The tests can be a bit unwieldy though because they use a lot of mocking and stuff, so if it's too much hassle, don't worry about it.

mazzi commented:
> Hey mazzi,
> 
> The node manager has a test suite. It would be nice if we could add a test case for this. The tests can be a bit unwieldy though because they use a lot of mocking and stuff, so if it's too much hassle, don't worry about it.

I've modified an existing test. I didn't wanted to copy past the one that I've modified.
Did you see it? It's alright?
Thanks!

jacderida commented:
> > Hey mazzi,
> > The node manager has a test suite. It would be nice if we could add a test case for this. The tests can be a bit unwieldy though because they use a lot of mocking and stuff, so if it's too much hassle, don't worry about it.
> 
> I've modified an existing test. I didn't wanted to copy past the one that I've modified. Did you see it? It's alright? Thanks!

Right, thanks. However, wouldn't it need an assertion that the assigned value is in lowercase?

mazzi commented:
> > > Hey mazzi,
> > > The node manager has a test suite. It would be nice if we could add a test case for this. The tests can be a bit unwieldy though because they use a lot of mocking and stuff, so if it's too much hassle, don't worry about it.
> > 
> > 
> > I've modified an existing test. I didn't wanted to copy past the one that I've modified. Did you see it? It's alright? Thanks!
> 
> Right, thanks. However, wouldn't it need an assertion that the assigned value is in lowercase?

The assertion was already there. I just modified the input. Please check the whole test.

jacderida commented:
> > > > Hey mazzi,
> > > > The node manager has a test suite. It would be nice if we could add a test case for this. The tests can be a bit unwieldy though because they use a lot of mocking and stuff, so if it's too much hassle, don't worry about it.
> > > 
> > > 
> > > I've modified an existing test. I didn't wanted to copy past the one that I've modified. Did you see it? It's alright? Thanks!
> > 
> > 
> > Right, thanks. However, wouldn't it need an assertion that the assigned value is in lowercase?
> 
> The assertion was already there. I just modified the input. Please check the whole test.

Ah ok cool, sorry. Approved.

mazzi committed:
Merge branch 'rc-2024.10.2' into fix_lowercase_owner
